client/src/components/location/Cities.js

```diff
@@ -24,14 +24,11 @@ class Cities extends Component {
   }
 
   render() {
-    let filteredCities = [];
-    const cities = this.props.cities || [];
-    const search = this.state;
+    let cities = this.props.cities || [];
+    const search = this.state.search;
 
-    filteredCities = cities;
-
-    filteredCities.filter(city => {
-      return city.name.toLowerCase().indexOf(search.search.toLowerCase) !== -1;
+    cities = cities.filter(city => {
```

Comment:
This was probably the problem, `Array.filter` doesn't mutate the existing Array, it just returns the filtered results.

Reply:
"cities" is read-only